bullet_train/docs/upgrades.md

```diff
@@ -24,47 +48,71 @@ git checkout .
 git clean -d -f
 ```
 
-### 2. Fetch the latest and greatest from the Bullet Train repository.
+### 3. Fetch the latest and greatest from the Bullet Train repository.
 
 ```
 git fetch bullet-train
+git fetch --tags bullet-train
 ```
 
-### 3. Create a new "upgrade" branch off of your main branch.
+### 4. Create a new "upgrade" branch off of your main branch.
+
+It can be handy to include the version number that you're moving to in the branch name.
 
 ```
 git checkout main
-git checkout -b updating-bullet-train
+git checkout -b updating-bullet-train-1.4.1
 ```
 
-### 4. Merge in the newest stuff from Bullet Train and resolve any merge conflicts.
+### 5. Merge in the newest stuff from Bullet Train and resolve any merge conflicts.
+
+Each version of the starter repo is tagged, so you can merge in the tag from the upstread repo.
 
 ```
-git merge bullet-train/main
+git merge v1.4.1
 ```
 
-It's quite possible you'll get some merge conflicts at this point. No big deal! Just go through and resolve them like you would if you were integrating code from another developer on your team. We tend to comment our code heavily, but if you have any questions about the code you're trying to understand, let us know on Discord!
+It's quite possible you'll get some merge conflicts at this point. No big deal! Just go through and
+resolve them like you would if you were integrating code from another developer on your team. We tend
+to comment our code heavily, but if you have any questions about the code you're trying to understand,
+let us know on Discord!
+
+One of the files that's likely to have conflicts, and which can be the most frustrating to resolve is
+`Gemfile.lock`. You can try to sort it out by hand, or you can checkout a clean copy and then let bundler
+generate a new one that matches what you need:
+
+```
+git checkout HEAD -- Gemfile.lock
+bundle update
+```
+
+If you choose to sort out `Gemfile.lock` by hand it's a good idea to run `bundle install` just to make
+sure that your `Gemfile.lock` agrees with the new state of `Gemfile`.
```

Comment:
For a different PR, but: iirc in `.gitattributes` we could assign a merge strategy that does this automatically?

Reply:
🤔 Oh, that's an interesting thought. Just speaking for myself personally, I'm not sure I'd want `.gitattributes` making that decision for me. Usually I want to take a look at `Gemfile.lock` to see how bad the conflicts are. If it's just a few gems I'll usually sort it out by hand and only turn to the `git checkout HEAD -- Gemfile.lock` if it's just to messy.

Reply:
Totally yeah, that works too! And we've got the suggestion in here now so people know about it.

Reply:
We could potentially mention something like "If you _always_ want to avoid conflicts in `Gemfile.lock` you can make that happy by doing XYZ to your `.gitattributes` file."